src/App.js

```diff
@@ -137,6 +141,7 @@ export default class App extends React.Component<{}, AppState> {
 
     handleCancelDashConnection = () => {
         this.setState({
+            modalIsShowing: false,
             showDashConnectionError: false
```

Comment:
Switch the order to:

    showDashConnectionError: false,
    modalIsShowing: false